lib/salus/scanners/sgrep.rb

```diff
@@ -0,0 +1,109 @@
+require "salus/scanners/base"
+require "json"
+
+# Report any matches to a list of semantic grep patterns provided by the config file.
+# We sgrep, a grep like tool for doing AST pattern matching. See https://sgrep.dev for more info and documentation.
+# Config file can provide and yaml block from the sgrep config. For example:
+#   - matches: Array[Hash]
+#       pattern:     <string>  (required)      pattern to match against.
+#       language:    <string>  (required) language the pattern is written in.
+#       forbidden:   <boolean> (default false) if true, a hit on this pattern will fail the test.
+#       required:    <boolean> (default false) if true, the absense of this pattern is a failure.
+#       message:     <string>  (default '')    custom message to display among failure.
+
+module Salus::Scanners
+  class Sgrep < Base
+    def run
+      # For each pattern, keep a running history of failures, errors, and hits
+      # These will be reported on at the end.
+      failure_messages = []
+      errors = []
+      all_hits = []
+
+      Dir.chdir(@repository.path_to_repo) do
+        base_path = Dir.pwd
+        @config["matches"]&.each do |match|
+          # sgrep has the following behavior:
+          #   always returns with status code 0 if able to parse
+          #   ouputs a json object with a 'results' key
+          #   this is either empty NO HITS
+          #   or had hit objects with a simple schema
+
+          # Set defaults.
+          match["forbidden"] ||= false
+          match["required"] ||= false
+          match["message"] ||= ""
+
+          # build the sgrep command string
+          command_string = [
+            "sgrep-lint",
```

Comment:
CC @pablo. Here's a good example of sgrep naming pain. Hoping we address this for good next week.